File: code_samples/typescript/algorithms_graph.ts
/**
 * Algorithms: Graph
 * AI/ML Training Sample
 */

interface IGraph {
    data: string;
    process(input: string): void;
    validate(): boolean;
}

class Graph implements IGraph {
    data: string;
    
    constructor() {
        this.data = "";
    }
    
    process(input: string): void {
        this.data = input;
    }
    
    getData(): string {
        return this.data;
    }
    
    validate(): boolean {
        return this.data.length > 0;
    }
}

// Example usage
const instance = new Graph();
instance.process("example");
console.log(`Data: ${instance.getData()}`);
console.log(`Valid: ${instance.validate()}`);

export { Graph, IGraph };

File: code_samples/r/algorithms_graph.r
# Algorithms: Graph
# AI/ML Training Sample

Graph <- setRefClass(
    "Graph",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- Graph$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")
